src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/DoNotIgnoreMethodResults.cs

```diff
@@ -177,7 +177,7 @@ public override void Initialize(AnalysisContext analysisContext)
                                 {
                                     rule = HResultOrErrorCodeRule;
                                 }
-                                else if (IsPureMethod(targetMethod, opContext.Compilation))
+                                else if (IsPureWithReturnMethod(targetMethod, opContext.Compilation))
```

Comment:
Instead of changing this method, can we change the check at start of this switch case `if (targetMethod == null)` to `if (targetMethod.ReturnsVoid)`?